src/neuroconv/tools/path_expansion.py

```diff
@@ -0,0 +1,77 @@
+import abc
+import os
+from pathlib import Path
+from typing import Iterable, List, Union
+
+from parse import parse
+from pydantic import DirectoryPath, FilePath
+
+from ..utils import DeepDict
+
+
+class AbstractPathExpander(abc.ABC):
+    def extract_metadata(self, folder, format_: str):
+        format_ = format_.replace("\\", os.sep)  # Actual character is a single back-slash; first is an escape for that
+        format_ = format_.replace("/", os.sep)  # our f-string uses '/' to communicate os-independent separators
+        for filepath in self.list_directory(folder):
+            result = parse(format_, filepath)
+            if result:
+                yield filepath, result.named
+
+    @abc.abstractmethod
+    def list_directory(self, folder: Union[FilePath, DirectoryPath]) -> Iterable[str]:
```

Comment:
@bendichter To make this argument more specific (as well as making the surrounding code more explicit about how it treats that variable) I was thinking about renaming it

1) `root`
2) `base`
3) `root_folder`
4) `base_folder`

Any preference?

Reply:
Or even

5. `root_directory` 
6. `base_directory`

to be consistent with the method name

Reply:
#6 was chosen